lib/databricks.js

```diff
@@ -24,6 +92,214 @@ export class DatabricksSingleton {
       );
     return DatabricksSingleton.instance;
   }
+
+  get types() {
+    if (DatabricksSingleton.types.size === 0)
+      throw new Error(
+        "DatabricksSingleton not initialized. Call DatabricksSingleton.initialize() first."
+      );
+    return DatabricksSingleton.types;
+  }
+}
+
+export async function describeTables(req, res, connection) {
+  const tablesData = {};
+  const session = await new Promise((resolve, reject) => {
+    connection.on("error", reject).openSession().then(resolve);
+  });
+
+  try {
+    const catalogOperation = await session.getCatalogs({runAsync: true});
+    const catalogs = await catalogOperation.fetchAll();
+
+    for (const catalog of catalogs) {
+      const tableOperation = await session.getTables({
+        catalogName: catalog.TABLE_CAT,
+        runAsync: true,
+      });
+      const tables = await tableOperation.fetchAll();
+      tables.forEach(
+        (table) =>
+          (tablesData[
+            `${table.TABLE_CAT}.${table.TABLE_SCHEM}.${table.TABLE_NAME}`
+          ] = table)
+      );
+    }
+  } catch (error) {
+    if (!error.statusCode) error.statusCode = 400;
+    throw error;
+  } finally {
+    if (session) {
+      try {
+        await session.close();
+      } catch (err) {
+        console.error(err);
+      }
+    }
+
+    if (connection) {
+      try {
+        await connection.close();
+      } catch (err) {
+        console.error(err);
+      }
+    }
+  }
+
+  return tablesData;
+}
+
+export async function describeColumns(req, res, connection) {
+  let columns = [];
+  const session = await new Promise((resolve, reject) => {
+    connection.on("error", reject).openSession().then(resolve);
+  });
+
+  const body = await json(req);
+
+  if (!validateDescribeColumnsPayload(body)) throw badRequest();
+
+  const {table, catalog, schema} = body;
```

Comment:
Currently when we query from the databases pane, we are only sending the table name (`{"table": "penguins"}`), and yet it successfully returns the columns. Any idea why that's working, despite not having catalog and schema?

Reply:
Note: we discussed this, makes sense.